library/ssl_tls13_client.c

```diff
@@ -2649,10 +2652,8 @@ static int ssl_encrypted_extensions_parse( mbedtls_ssl_context* ssl,
 
     while( ext_len )
     {
-        unsigned int ext_id = ( ( ext[0] << 8 )
-                                | ( ext[1] ) );
-        unsigned int ext_size = ( ( ext[2] << 8 )
-                                  | ( ext[3] ) );
+        unsigned int ext_id = ( ( ext[0] << 8 ) | ( ext[1] ) );
+        unsigned int ext_size = ( ( ext[2] << 8 ) | ( ext[3] ) );
```

Comment:
`ext_size` can be `size_t`. Then it is the same type as `ext_len` and there is no need for the cast on L2690. Also it is not clear to me why there is no cast on L2676.

Reply:
Yeah, I have this observation too. However, [many places](https://github.com/hannestschofenig/mbedtls/blob/tls13-prototype/library/ssl_cli.c#L2137) in TLS 1.2 uses `unsigned int` for the `ext_size`.

Reply:
Maybe the cast on L2690 is not needed?

Reply:
@lhuang04 is right that there are other places where we use `unsigned int`, but I think those should be considered bugs, and we generally aim to use `size_t` for every variable holding a buffer length. I'd therefore suggest to change the type of `ext_size` to `size_t`.